### PR TITLE
[binder-plugin] Add option to disable verbose logging completely. JB#52016

### PR DESCRIPTION
**Pull request by ikamaletdinov**
It is possible to disable all verbose logging during compile time for
nfcd-binder-plugin adding additional rpm macro %{disable_hexdump}
during build.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -49,7 +49,7 @@ CC = $(CROSS_COMPILE)gcc
 LD = $(CC)
 WARNINGS = -Wall
 BASE_FLAGS = -fPIC -fvisibility=hidden
-DEFINES = -DNFC_PLUGIN_EXTERNAL
+override DEFINES := -DNFC_PLUGIN_EXTERNAL $(DEFINES)
```
Comment (monich) [inline Makefile]: DEFINES += -DNFC_PLUGIN_EXTERNAL

**Comment by monich** [comment]
Let's use DISABLE_HEXDUMP instead of `_NO_NFC_VERBOSE_`. "if not disable hexdump" is easier to comprehend than "if not no nfc verbose"

File: src/binder_nfc_adapter.c [hunk 2]
```diff
@@ -46,7 +46,11 @@
 GLogModule binder_hexdump_log = {
     .name = "binder-hexdump",
     .parent = &GLOG_MODULE_NAME,
+#ifndef _NO_NFC_VERBOSE_
     .max_level = GLOG_LEVEL_MAX,
+#else
+    .max_level = GLOG_LEVEL_DEBUG,
+#endif /* _NO_NFC_VERBOSE_ */
```
Comment (monich) [inline src/binder_nfc_adapter.c]: I don't think this is necessary

File: src/binder_nfc_adapter.c [hunk 3]
```diff
@@ -242,8 +246,10 @@ binder_nfc_callback_handle_data(
     if (data && gbinder_reader_at_end(reader)) {
         NciHalClient* hal_client = self->hal_client;
 
-        DUMP("%c data, %u byte(s)", DIR_IN, (guint)len);
+#ifndef _NO_NFC_VERBOSE_
+        gutil_log(&binder_hexdump_log, GLOG_LEVEL_VERBOSE, "%c data, %u byte(s)", DIR_IN, (guint)len);
         binder_dump_data(DIR_IN, data, len);
+#endif /* _NO_NFC_VERBOSE_ */
```
Comment (monich) [inline src/binder_nfc_adapter.c]: Please define one macro (e.g. `DUMP()`) at the top and use it

File: src/binder_nfc_adapter.c [hunk 4]
```diff
@@ -328,7 +334,9 @@ binder_nfc_client_write(
     GBinderWriter writer;
     gulong id;
 
+#ifndef _NO_NFC_VERBOSE_
     binder_dump_data(DIR_OUT, data, len);
+#endif /* _NO_NFC_VERBOSE_ */
```
Comment (monich) [inline src/binder_nfc_adapter.c]: Instead of having `#ifndef`s all over the place.

File: src/binder_nfc_adapter.c (hunk 2 above)
Comment (ikamaletdinov) [reply src/binder_nfc_adapter.c]: Thanks for the review. Agreed. Removed.

File: src/binder_nfc_adapter.c (hunk 3 above)
Comment (ikamaletdinov) [reply src/binder_nfc_adapter.c]: Thanks for the review. I have combined that parts together.

File: Makefile (hunk 1 above)
Comment (ikamaletdinov) [reply Makefile]: Thanks for the review. That part was reworked.

**Review by monich — APPROVED** [review]
LGTM

**Pull request merged**